providers/userdetail/src/java/org/sakaiproject/user/detail/ValueEncryptionService.java

```diff
@@ -0,0 +1,119 @@
+package org.sakaiproject.user.detail;
+
+import java.security.AlgorithmParameters;
+import java.security.SecureRandom;
+import java.security.spec.KeySpec;
+import java.util.Base64;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.sakaiproject.component.cover.ServerConfigurationService;
+
+/**
+ * This class provides encryption/decryption services. Service is thread safe.
+ */
+public class ValueEncryptionService {
```

Comment:
At the beginning, it was a service and we move it to a utility because, as you suggested, it only was used by our provider implementations, and has no sense to make it accessible for the whole Sakai